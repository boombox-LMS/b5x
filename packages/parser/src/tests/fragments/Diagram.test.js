const {
  exampleDiagramMarkup,
  DiagramDataSchema,
} = require("../../../dist/componentBuilders/fragments/parsingClasses/Diagram");
const { RawFragmentSchema } = require("@b5x/types");

describe("The Diagram fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(exampleDiagramMarkup);
  const rawFragment = fragment.packageForApi();

  test("The parsed diagram fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed diagram fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });

  test("The parsed diagram fragment uses the expected data schema", () => {
    const validator = () => {
      DiagramDataSchema.parse(rawFragment.data);
    };
    expect(validator).not.toThrowError();
  });
});
