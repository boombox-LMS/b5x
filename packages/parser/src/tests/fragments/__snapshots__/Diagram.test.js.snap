// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Diagram fragment builds as expected The parsed diagram fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [],
  "contentType": "Diagram",
  "contents": "",
  "data": Object {
    "elements": Array [
      Object {
        "classes": Array [],
        "data": Object {
          "col": "1",
          "id": "client",
          "row": "2",
        },
        "group": "nodes",
      },
      Object {
        "classes": Array [
          "node--icon",
        ],
        "data": Object {
          "col": "1",
          "id": "client-content",
          "label": "Client",
          "parent": "client",
          "row": "2",
        },
        "group": "nodes",
      },
      Object {
        "classes": Array [],
        "data": Object {
          "col": "3",
          "id": "lb",
          "row": "2",
        },
        "group": "nodes",
      },
      Object {
        "classes": Array [
          "node--icon",
        ],
        "data": Object {
          "col": "3",
          "id": "lb-content",
          "label": "Load balancer",
          "parent": "lb",
          "row": "2",
        },
        "group": "nodes",
      },
      Object {
        "classes": Array [],
        "data": Object {
          "col": "5",
          "id": "server1",
          "row": "1",
        },
        "group": "nodes",
      },
      Object {
        "classes": Array [
          "node--icon",
        ],
        "data": Object {
          "col": "5",
          "id": "server1-content",
          "label": "App server",
          "parent": "server1",
          "row": "1",
        },
        "group": "nodes",
      },
      Object {
        "classes": Array [],
        "data": Object {
          "col": "5",
          "id": "server2",
          "row": "2",
        },
        "group": "nodes",
      },
      Object {
        "classes": Array [
          "node--icon",
        ],
        "data": Object {
          "col": "5",
          "id": "server2-content",
          "label": "App server",
          "parent": "server2",
          "row": "2",
        },
        "group": "nodes",
      },
      Object {
        "classes": Array [],
        "data": Object {
          "col": "5",
          "id": "server3",
          "row": "3",
        },
        "group": "nodes",
      },
      Object {
        "classes": Array [
          "node--icon",
        ],
        "data": Object {
          "col": "5",
          "id": "server3-content",
          "label": "App server",
          "parent": "server3",
          "row": "3",
        },
        "group": "nodes",
      },
      Object {
        "classes": Array [],
        "data": Object {
          "col": "7",
          "id": "db",
          "row": "2",
        },
        "group": "nodes",
      },
      Object {
        "classes": Array [
          "node--icon",
        ],
        "data": Object {
          "col": "7",
          "id": "db-content",
          "label": "Database",
          "parent": "db",
          "row": "2",
        },
        "group": "nodes",
      },
      Object {
        "classes": Array [
          "edge--source-arrow",
          "edge--target-arrow",
          "edge--horizontal",
        ],
        "data": Object {
          "id": "client-lb-http",
          "label": "http",
          "source": "client",
          "target": "lb",
        },
        "group": "edges",
      },
      Object {
        "classes": Array [
          "edge--source-arrow",
          "edge--target-arrow",
        ],
        "data": Object {
          "id": "server1-lb-http",
          "label": "http",
          "source": "server1",
          "target": "lb",
        },
        "group": "edges",
      },
      Object {
        "classes": Array [
          "edge--source-arrow",
          "edge--target-arrow",
          "edge--horizontal",
        ],
        "data": Object {
          "id": "server2-lb-http",
          "label": "http",
          "source": "server2",
          "target": "lb",
        },
        "group": "edges",
      },
      Object {
        "classes": Array [
          "edge--source-arrow",
          "edge--target-arrow",
        ],
        "data": Object {
          "id": "server3-lb-http",
          "label": "http",
          "source": "server3",
          "target": "lb",
        },
        "group": "edges",
      },
      Object {
        "classes": Array [
          "edge--source-arrow",
          "edge--target-arrow",
        ],
        "data": Object {
          "id": "server1-db-http",
          "label": "http",
          "source": "server1",
          "target": "db",
        },
        "group": "edges",
      },
      Object {
        "classes": Array [
          "edge--source-arrow",
          "edge--target-arrow",
          "edge--horizontal",
        ],
        "data": Object {
          "id": "server2-db-http",
          "label": "http",
          "source": "server2",
          "target": "db",
        },
        "group": "edges",
      },
      Object {
        "classes": Array [
          "edge--source-arrow",
          "edge--target-arrow",
        ],
        "data": Object {
          "id": "server3-db-http",
          "label": "http",
          "source": "server3",
          "target": "db",
        },
        "group": "edges",
      },
    ],
    "stylesheet": Array [
      Object {
        "selector": "#client-content",
        "style": Object {
          "backgroundFit": "contain",
          "backgroundImage": "https://api.iconify.design/ic/baseline-phone-android.svg",
        },
      },
      Object {
        "selector": "#lb-content",
        "style": Object {
          "backgroundFit": "contain",
          "backgroundImage": "https://api.iconify.design/carbon/network-2.svg",
        },
      },
      Object {
        "selector": "#server1-content",
        "style": Object {
          "backgroundFit": "contain",
          "backgroundImage": "https://api.iconify.design/mingcute/server-line.svg",
        },
      },
      Object {
        "selector": "#server2-content",
        "style": Object {
          "backgroundFit": "contain",
          "backgroundImage": "https://api.iconify.design/mingcute/server-line.svg",
        },
      },
      Object {
        "selector": "#server3-content",
        "style": Object {
          "backgroundFit": "contain",
          "backgroundImage": "https://api.iconify.design/mingcute/server-line.svg",
        },
      },
      Object {
        "selector": "#db-content",
        "style": Object {
          "backgroundFit": "contain",
          "backgroundImage": "https://api.iconify.design/mdi/database-outline.svg",
        },
      },
    ],
  },
  "dependencyUris": Array [],
  "digest": "d820d2511d932c89f4a9a63d104d9640",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": false,
  "isStateful": true,
  "uri": "test-fragment-uri",
}
`;
